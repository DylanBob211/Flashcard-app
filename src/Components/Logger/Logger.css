select {
    -webkit-appearance: none;
    -moz-appearance: none;
    appearance: none;
}


.loginCard_container{
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    display: flex;
    flex-direction: column;
    background: var(--primary-color);
    border-radius: 10px;
    padding: 2rem;
    color: #fff;
}

.loginCard_welcome {
    text-align: center;
    font-size: 4rem;
}

.loginCard_form{
    display: flex;
    justify-content: center;
    align-items: center;
}

.loginCard_select{
    width: 8em;
    height: 2em;    
    border: 1px solid var(--second-shade);
    border-radius: 5px;
    font-size: 1rem;
    text-align: center;
    background: var(--third-shade-secondary);
    transition: all .2s;
    font-size: 1rem;
    font-weight: 800;
}

.loginCard_select option {
    font-size: 1rem;
    font-weight: 800;
}

.loginCard_select:focus{
    outline: none;
    background: var(--secondary-color);
}

.loginCard_select:hover{
    background: var(--first-shade-secondary);
}

.loginCard_button{
    height: 3em;
    width: 6em;
    cursor: pointer;
    margin-left: 2em;
    border-radius: 5px;
    
    padding: .6em;
    transition: all .2s;
    font-weight: 800;
    font-size: 1.5rem;
}

.loginCard_button--primary{
    background: var(--third-shade-secondary);
}

.loginCard_button--primary:hover{
    background: var(--secondary-color);
}

.loginCard_button--secondary{
    background: var(--clashing-color);
}

.loginCard_button--secondary:hover{
    background: var(--second-shade-clashing)
}