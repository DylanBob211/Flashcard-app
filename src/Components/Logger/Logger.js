import React, {useEffect, useState} from 'react'
import { Redirect } from 'react-router-dom'


const Logger = props => {

    return (
        <div>
            <h1>Benvenuto nella mia flashcard App</h1>
            <h2>clicca sul pulsante per continuare</h2>
            <button onClick={e => window.location.assign('/main')}>click to continue</button>
        </div>
    )
}

export default Logger
