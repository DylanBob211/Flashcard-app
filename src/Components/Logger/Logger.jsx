import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import './Logger.css';


const Logger = () => {
  const [step, setStep] = useState(1);

  const nextStep = () => setStep(() => step + 1);

  const prevStep = () => setStep(() => step - 1);

  switch (step) {
    case 1: return <FirstCard next={nextStep} />;
    case 2: return <SecondCard prev={prevStep} />;
    default: return <FirstCard />;
  }
};

export default Logger;

const FirstCard = ({ next }) => (
  <div className="loginCard_container">
    <h1 className="loginCard_welcome">Welcome</h1>
    <h2 className="loginCard_order">Choose your mother tongue</h2>
    <form form className="loginCard_form">
      <select className="loginCard_select">
        <option>bla</option>
        <option>qua</option>
        <option>dia</option>
      </select>
      <button
        className="loginCard_button loginCard_button--primary"
        onClick={e => next()}
      >
          Continue
      </button>
    </form>
  </div>
);

FirstCard.propTypes = {
  next: PropTypes.func.isRequired,
};

const SecondCard = ({ prev }) => {
  const goAhead = (e) => {
    e.preventDefault();
    // TODO: submit result
    window.location.assign('/main');
  };
  return (
    <div className="loginCard_container">
      <h2 className="loginCard_order">Choose the language that you want to learn</h2>
      <form className="loginCard_form">
        <select className="loginCard_select">
          <option>bla</option>
          <option>qua</option>
          <option>dia</option>
        </select>
        <button className="loginCard_button loginCard_button--secondary" onClick={e => prev()}>Go back</button>
        <button className="loginCard_button loginCard_button--primary" onClick={e => goAhead(e)} type="submit">Continue</button>
      </form>
    </div>
  );
};

SecondCard.propTypes = {
  prev: PropTypes.func.isRequired,
};