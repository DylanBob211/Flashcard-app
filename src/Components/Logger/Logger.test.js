import React from 'react';
import { mount, shallow } from 'enzyme';
import { Redirect } from 'react-router-dom';
import Logger from './Logger';

jest.mock('react-router-dom');

describe('Logger', () => {
  let mountedLoggerComponent;
  const props = {
    languages: {
      from: '',
      to: '',
    },
    setLanguages: jest.fn(),
  };

  const logger = ({ languages, setLanguages }, isShallow = false) => {
    if (isShallow) {
      return shallow(
        <Logger
          languages={languages}
          setLanguages={setLanguages}
        />,
      );
    }
    return mount(
      <Logger
        languages={languages}
        setLanguages={setLanguages}
      />,
    );
  };

  beforeEach(() => {
    jest.clearAllMocks();
    mountedLoggerComponent = undefined;
  });

  describe('Logger Component With First Card Rendered', () => {
    beforeEach(() => {
      mountedLoggerComponent = logger(props);
    });

    it('Matches to snapshot', () => {
      expect(mountedLoggerComponent).toMatchSnapshot();
    });

    it('renders FirstCard when mounted, not SecondCard', () => {
      const firstCard = mountedLoggerComponent
        .find('[data-test="firstCardWrapper"]');
      const secondCard = mountedLoggerComponent
        .find('[data-test="secondCardWrapper"]');

      expect(firstCard.exists()).toEqual(true);
      expect(secondCard.exists()).toEqual(false);
    });

    it('calls setLanguages when selecting a language', () => {
      const selectLanguage = mountedLoggerComponent
        .find('[data-test="firstLoggerScreenLanguageSelection"]');
      selectLanguage.simulate('change', { target: { value: 'en' } });
      expect(props.setLanguages).toHaveBeenCalled();
    });

    it('cannot press the button Next if no value for prop languages.from is selected', () => {
      const buttonNext = mountedLoggerComponent
        .find('[data-test="firstLoggerScreenNextButton"]');
      const secondCard = mountedLoggerComponent
        .find('[data-test="secondCardWrapper"]');
      buttonNext.simulate('click');
      expect(buttonNext.props().disabled).toEqual(true);
      expect(secondCard.exists()).toEqual(false);
    });
  });

  describe('Logger Component with "from" props inputted', () => {
    beforeEach(() => {
      props.languages.from = 'en';
      mountedLoggerComponent = logger(props);
    });
    it('enables the Next button to be pressed', () => {
      const buttonNext = mountedLoggerComponent
        .find('[data-test="firstLoggerScreenNextButton"]');
      expect(buttonNext.props().disabled).toEqual(false);
    });
  });

  describe('Logger Component with Second Card Rendering', () => {
    beforeEach(() => {
      props.languages.from = 'en';
      mountedLoggerComponent = logger(props);
      const buttonNext = mountedLoggerComponent
        .find('[data-test="firstLoggerScreenNextButton"]');
      buttonNext.simulate('click');
    });

    it('renders SecondCard, not FirstCard anymore', () => {
      const secondCard = mountedLoggerComponent
        .find('[data-test="secondCardWrapper"]');
      const firstCard = mountedLoggerComponent
        .find('[data-test="firstCardWrapper"]');
      expect(secondCard.exists()).toEqual(true);
      expect(firstCard.exists()).toEqual(false);
    });

    it('disables Start! button', () => {
      const startButton = mountedLoggerComponent
        .find('[data-test="secondLoggerScreenNextButton"]');
      expect(startButton.props().disabled).toEqual(true);
    });

    it('goes back to the first card if "Back" button is pressed', () => {
      const secondCard = mountedLoggerComponent
        .find('[data-test="secondCardWrapper"]');
      const backButton = mountedLoggerComponent
        .find('[data-test="secondLoggerScreenBackButton"]');
      expect(secondCard.exists()).toEqual(true);
      backButton.simulate('click');
      const firstCard = mountedLoggerComponent
        .find('[data-test="firstCardWrapper"]');
      expect(firstCard.exists()).toEqual(true);
    });

    it('cannot press the button if no language for the "to" key has been inserted', () => {
      const secondCard = mountedLoggerComponent
        .find('[data-test="secondCardWrapper"]');
      const nextButton = mountedLoggerComponent
        .find('[data-test="secondLoggerScreenNextButton"]');
      nextButton.simulate('click');
      expect(secondCard.exists()).toEqual(true);
    });

    it('calls setLanguages when selecting a language', () => {
      const selectLanguage = mountedLoggerComponent
        .find('[data-test="secondLoggerScreenLanguageSelection"]');
      selectLanguage.simulate('change', { target: { value: 'en' } });
      expect(props.setLanguages).toHaveBeenCalled();
    });
  });

  describe('Logger SecondCard with language props inserted', () => {
    beforeEach(() => {
      props.languages = { from: 'en', to: 'it' };
      mountedLoggerComponent = logger(props);
      const buttonNext = mountedLoggerComponent
        .find('[data-test="firstLoggerScreenNextButton"]');
      buttonNext.simulate('click');
    });

    it('enables Start! button', () => {
      const startButton = mountedLoggerComponent
        .find('[data-test="secondLoggerScreenNextButton"]');
      expect(startButton.props().disabled).toEqual(false);
    });

    it('renders Redirect component when Start! button is pressed', () => {
      Redirect.mockReturnValue(<div data-test="redirect" />);
      const startButton = mountedLoggerComponent
        .find('[data-test="secondLoggerScreenNextButton"]');
      startButton.simulate('click');

      expect(mountedLoggerComponent.contains(<div data-test="redirect" />)).toEqual(true);
    });
  });
});
