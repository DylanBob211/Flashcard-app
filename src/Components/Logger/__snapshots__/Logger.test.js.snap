// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logger Logger Component With First Card Rendered Matches to snapshot 1`] = `
<Logger
  languages={
    Object {
      "from": "",
      "to": "",
    }
  }
  setLanguages={[MockFunction]}
>
  <div
    className="loginCard_container"
    data-test="firstCardWrapper"
  >
    <h1
      className="loginCard_welcome"
    >
      Welcome
    </h1>
    <h2
      className="loginCard_order"
    >
      Choose your mother tongue
    </h2>
    <div
      className="loginCard_form"
    >
      <select
        className="loginCard_select"
        data-test="firstLoggerScreenLanguageSelection"
        name="from"
        onChange={[Function]}
        value=""
      >
        <option
          disabled={true}
          value=""
        />
        <option
          disabled={false}
          value="en"
        >
          English
        </option>
        <option
          disabled={false}
          value="it"
        >
          Italian
        </option>
        <option
          disabled={false}
          value="ru"
        >
          Russian
        </option>
      </select>
      <button
        className="loginCard_button loginCard_button--primary"
        data-test="firstLoggerScreenNextButton"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Continue
      </button>
    </div>
  </div>
</Logger>
`;
